### PR TITLE
[pjrt] Add ROCM and Vulkan backends.

### PR DESCRIPTION
**Pull request by stellaraccident**
* The vulkan backend raises an error on some memory transfer thing that needs some debugging for d/h transfers.
* I don't have a functioning ROCM installation on this machine but it compiles and loads to the point of trying to find ROCM libraries at runtime.
* I'm going to raise an RFC to enable ROCM compiler backend support in default builds as then ROCM would work out of the box with shipped project compiler binary packages.

### CODE REVIEW

File: integrations/pjrt/python_packages/iree_rocm_plugin/setup.py
Comment (raikonenfnu) [inline integrations/pjrt/python_packages/iree_rocm_plugin/setup.py:35]: Thanks Stella for this cool addition! Overall rocm bit looks good to me!

Comment (stellaraccident) [reply integrations/pjrt/python_packages/iree_rocm_plugin/setup.py:35]: Go ahead and have a once over of all of it and approve if looks good. None of this is in CI yet, and we have a couple more iterations before usable generally...

File: integrations/pjrt/python_packages/iree_rocm_plugin/jax_plugins/iree_rocm/__init__.py
Comment (raikonenfnu) [inline integrations/pjrt/python_packages/iree_rocm_plugin/jax_plugins/iree_rocm/__init__.py:37]: QQ: Any reasons we are setting all the priorities to 500? Would it default to other jax backend if we don't do this? https://jax.readthedocs.io/en/latest/_modules/jax/_src/xla_bridge.html

Comment (stellaraccident) [reply integrations/pjrt/python_packages/iree_rocm_plugin/jax_plugins/iree_rocm/__init__.py:37]: copy-paste for the moment and "it works". We'll change them all at once if there is other guidance.

Comment (raikonenfnu) [reply integrations/pjrt/python_packages/iree_rocm_plugin/jax_plugins/iree_rocm/__init__.py:37]: Understood, LGTM!

**Review by jpienaar — APPROVED** [review]
Overall looks good! (initially I missed PJRT in the title and thought this was doing something different and was going to ask about moving PJRT changes out  to their own PR :))

**Review by raikonenfnu — APPROVED** [review]

**Pull request merged**
